src/libraries/System.Runtime/ref/System.Runtime.cs

```diff
@@ -6838,6 +6838,7 @@ public abstract partial class Type : System.Reflection.MemberInfo, System.Reflec
         public virtual System.Type MakeByRefType() { throw null; }
         public static System.Type MakeGenericMethodParameter(int position) { throw null; }
         public static System.Type MakeGenericSignatureType(System.Type genericTypeDefinition, params System.Type[] typeArguments) { throw null; }
+        [System.Diagnostics.CodeAnalysis.RequiresUnreferencedCodeAttribute("If some of the generic arguments are annotated (either with DynamicallyAccessedMembersAttribute, or generic constraints), trimming can't validate that the requirements of those annotations are met.")]
         public virtual System.Type MakeGenericType(params System.Type[] typeArguments) { throw null; }
```

Comment:
I added some support in the linker for validation of array parameters specifically to enable analyzing `MakeGenericType` effectively for arrays with known types passed in and a known `this` value. Why is this being marked as `[RequiresUnreferencedCode]` now?

Reply:
With this annotation, C#/WinRT effectively becomes untrimmable because there will be `[RequiresUnreferencedCode]` warnings everywhere.

Reply:
The linker cannot fully reason about annotations on generic argument types and hence the warnings. Regarding C#/WinRT trimming, are there any reasonable workarounds like suppressing the warnings in an easy way?

Reply:
The linker can reason about the annotations on the generic arguments. I specifically tested that when I did the array work.

For C#/WinRT, it could suppress the warnings, but then it would need to manually propagate the annotations, which is difficult when the linker tooling doesn't help (or actively makes the problem worse, which this will).

Reply:
I added the support for recognizing the reflection pattern in https://github.com/mono/linker/pull/1955

Reply:
See a long discussion on this PR, #55102, on the current in linker limitation in .NET6 on open generic analysis

Reply:
I believe the intention is for this to warn only when someone tries to access `MakeGenericType` via reflection since they could invoke it with unknown inputs that we can't validate. We should double-check that the annotation doesn't cause warnings when it's called with known inputs.

Reply:
The attribute annotations are "the worst case" - linker is free to not warn if it knows better (same for analyzers and so on). In this case the annotation is there for cases where we don't have intrinsic handling. Some of these are in the linker, but any other tool might be in the same position. We have to basically "play it safe".

Reply:
As long as the linker/analyzer won't warn in cases where all of the inputs are known and can be statically analyzed, then I'll be happy.